tensorflow/src/main/scala/magnolify/tensorflow/ExampleType.scala

```diff
@@ -119,30 +120,23 @@ object ExampleField {
         f
       }
 
-    override protected def buildSchema(k: String)(cm: CaseMapper): Schema = {
-      val sb = Schema
-        .newBuilder()
+    override protected def buildSchema(cm: CaseMapper): Schema = {
+      val sb = Schema.newBuilder()
       getDoc(caseClass.annotations, caseClass.typeName.full).foreach(sb.setAnnotation)
-
       caseClass.parameters.foldLeft(sb) { (b, p) =>
-        val s = p.typeclass.buildSchema(key(k, cm.map(p.label)))(cm)
-        val annotatedFs = s.getFeatureList.asScala
-          .map { f =>
-            val maybeAnnotation = if (f.hasAnnotation) {
-              Some(f.getAnnotation)
-            } else {
-              getDoc(p.annotations, s"${caseClass.typeName.full}#${key(k, cm.map(p.label))}")
-            }
-
-            maybeAnnotation match {
-              case Some(a) => f.toBuilder.setAnnotation(a).build()
-              case None    => f
-            }
-          }
-        b.addAllFeature(annotatedFs.asJava)
+        val n = cm.map(p.label)
+        val s = p.typeclass.schema(cm)
+        val fs = s.getFeatureList.asScala.map { f =>
+          val fb = f.toBuilder
+          val fk = if (f.hasName) key(n, f.getName) else n // primitives do not have name
+          val d = getDoc(p.annotations, s"${caseClass.typeName.full}#$fk")
+          fb.setName(fk)
+          if (!f.hasAnnotation) d.foreach(fb.setAnnotation)
```

Comment:
This seems like the reverse of what you want to do?

Reply:
In the original code we have `maybeAnnotation` which is the annotation from `f` if any, or else the annotation from `p` with `getDoc`. We then set it on the builder.

Here, instead of setting again the annotation on `f` (the builder already contains it since derived from `f`), I only add the `p` annotation if there are none from `f`